boto3/session.py

```diff
@@ -72,6 +78,13 @@ def __repr__(self):
         return 'Session(region={0})'.format(
             repr(self._session.get_config_variable('region')))
 
+    @property
+    def profile_name(self):
+        """
+        The **read-only** profile name.
+        """
+        return self._session.profile or 'default'
```

Comment:
I would strongly prefer that this always return a string, but it could be either `'default'` or a blank string. It makes code like this simpler:

```python
if session.profile.startswith('prod'):
    # Do something here!
    pass
```